app/src/main/java/com/keylesspalace/tusky/components/account/AccountViewModel.kt

```diff
@@ -316,7 +310,7 @@ class AccountViewModel @Inject constructor(
     }
 
     private fun reload(isReload: Boolean = false) {
-        if (isDataLoading) {
+        if (_isRefreshing.value) {
```

Comment:
Two loads may still be triggered in parallel during the initial load, because `_isRefreshing` is not set to true in that case.

Hence I suggest to keep the loading state in a separate field, or merge loading and refreshing states together using a sealed class.

Alternatively, cancel the load in progress before starting a new one.